course_catalog/etl/mitpe.py

```diff
@@ -0,0 +1,212 @@
+        "div", {"class": "field--type-text-with-summary"}
+    )
+    # pylint: disable=expression-not-assigned
+    [p.extract() for p in div.findAll("p", {"class": "special"})]
+    paragraphs = []
+    for child in div.findAll():
+        if child.name == "p":
+            paragraphs.append(child.text)
+        elif child.name == "h3":
+            break
+
+    return paragraphs
+
+
+@log_exceptions(
+    "Error extracting MIT Professional Education catalog", exc_return_value=[]
+)
+def extract():
+    """Loads the MIT Professional Education catalog data via BeautifulSoup"""
+    if not settings.MITPE_BASE_URL:
+        log.error("MIT Professional base URL not set, skipping ETL")
+        return []
+    courses = []
+    soup = bs(
+        requests.get(urljoin(settings.MITPE_BASE_URL, "/course-catalog")).content,
+        "html.parser",
+    )
+    listings = soup.find("div", {"class": "course-listing"}).findAll(
+        "div", {"class": "views-row"}
+    )
+    for listing in listings:
+        link = listing.find("a")
+        url = urljoin(settings.MITPE_BASE_URL, link.get("href"))
+        details = bs(requests.get(url).content, "html.parser")
+        description = _parse_description(details)
+        short_description = strip_extra_whitespace(description[0])
+        full_description = strip_extra_whitespace(" ".join(p for p in description))
+
+        courses.append(
+            {
+                "url": url,
+                "title": strip_extra_whitespace(link.get_text()),
+                "dates": _parse_run_dates(listing),
+                "price": _parse_price(listing),
+                "topics": _parse_topics(listing),
+                "instructors": _parse_instructors(listing),
+                "short_description": short_description,
+                "full_description": full_description,
+            }
+        )
+    return courses
+
+
+@log_exceptions(
+    "Error transforming MIT Professional Education catalog", exc_return_value=[]
+)
+def transform(courses):
+    """Transform the MIT Professional Education course data"""
+    return [
+        {
+            "url": course["url"],
+            "title": course["title"],
+            "topics": [{"name": topic} for topic in course["topics"]],
+            "short_description": course["short_description"],
+            "full_description": course["full_description"],
+            "course_id": generate_unique_id(course["url"]),
+            "platform": PlatformType.mitpe.value,
+            "offered_by": [{"name": OfferedBy.mitpe.value}],
+            "runs": [
+                {
+                    "url": course["url"],
+                    "prices": ([{"price": course["price"]}] if course["price"] else []),
+                    "run_id": generate_unique_id(
+                        f"{course['url']}{datetime.strftime(date_range[0], '%Y%m%d')}"
+                    ),
+                    "platform": PlatformType.mitpe.value,
+                    "start_date": date_range[0],
```

Comment:
We probably don't care but the day before the start date is displayed as the start date in the drawer due to UTC conversion issues

Reply:
I care. Can we set the start time to something arbitrary, but likely, like 9am EST ?

Reply:
Good catch.  Changed to a default of noon UTC (== 7-8 Eastern, 4-5 Pacific)